common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordReadMarshallable.java

```diff
@@ -80,23 +83,41 @@ private StoredAuditRecord readV1(WireIn wire)
         checkV1Type(wire);
         int bitmap = wire.read(WireTags.KEY_FIELDS).int32();
 
-        FieldSelector fields = FieldSelector.fromBitmap(bitmap);
-        StoredAuditRecord.Builder recordBuilder = StoredAuditRecord.builder();
+        FieldSelector fieldsV1 = FieldSelector.fromBitmap(bitmap);
```

Comment:
I don't think we really need to break this out into 2 separate methods...  (or maybe just for being more clear?)
But the only reason for needing a new version as this line:
FieldSelector fieldsV1 = FieldSelector.fromBitmap(bitmap);
That it could throw if an old version of the code tried to read/parse a new chronicle entry -> that could cause FieldSelector.fromBitmap to throw...
BUT here we are "guarded" from this by the version-check in the readMarshallable-method... 
So that should be no issue coming here...
So I guess it should be enough to add the call:
  fieldsV2.ifSelectedRun(Field.SUBJECT, () -> recordBuilder.withSubject(wire.read(WireTags.KEY_SUBJECT).text()));
to the existing readV1-method...

BUT in case we keep one method (readV1 == readV2) maybe we should rename it to something else...
What differ from readV0 is that
V0 - reads type == ecaudit-batch && type == ecaudit-single
V1/2 - reads type == ecaudit

Reply:
Good points, I'll simplify back to one method